packages/sdk-ui-ts/src/services/ibc/tokens.ts

```diff
@@ -49,9 +49,24 @@ function ibcTokenMetadataToToken(
   })
 }
 
+function timeout(ms: number) {
+  return new Promise((_, reject) =>
+    setTimeout(() => reject(new Error('Request timed out')), ms),
+  )
+}
+
 ;(async () => {
   try {
-    const response = (await ibcTokenMetadataApi.get(TOKEN_METADATA_PATH)) as {
+    const ibcTokenMetadataResponse = ibcTokenMetadataApi.get(
+      TOKEN_METADATA_PATH,
+    ) as Promise<{
+      data: IbcTokenMetadata[]
+    }>
+
+    const response = (await Promise.race([
+      ibcTokenMetadataResponse,
+      timeout(2000),
```

Comment:
if response from ibc token metadata fetch takes longer than `2` seconds, the `timeout` function will throw an error, which will just get logged in the `catch` block below. then, build should continue

Reply:
We can pass second parameter to HttpRestClient { timeout: 2000 } and have the same logic as you implemented